lgc/patch/MeshTaskShader.cpp

```diff
@@ -2165,6 +2057,61 @@ void MeshTaskShader::collectMeshStatsInfo(Function *entryPoint, Value *numMeshPr
   }
 }
 
+// =====================================================================================================================
+// Do exporting. The array of values for certain export kind are all exported.
+//
+// @param kind : Export kind (positions, primitive, or parameters)
+// @param exports : Array of exports
+void MeshTaskShader::doExport(ExportKind kind, ArrayRef<ExportInfo> exports) {
+  for (unsigned i = 0; i < exports.size(); ++i) {
+    auto &values = exports[i].values;
+
+    assert(values[0]); // Must at least have one value
+    auto valueTy = values[0]->getType();
+    assert(valueTy->isFloatTy() || valueTy->isIntegerTy(32)); // Must be float or i32
+
+    auto undef = PoisonValue::get(valueTy);
+    unsigned numValues = 0;
+    for (auto value : values) {
+      if (value)
+        ++numValues;
+    }
+
+    unsigned target = InvalidValue;
+    switch (kind) {
+    case ExportKind::Pos:
+      target = EXP_TARGET_POS_0;
+      break;
+    case ExportKind::Prim:
+      target = EXP_TARGET_PRIM;
+      break;
+    case ExportKind::VertAttr:
+    case ExportKind::PrimAttr:
+      target = EXP_TARGET_PARAM_0;
+      break;
+    default:
+      llvm_unreachable("Unexpected export target!");
+      break;
+    }
+
+    bool exportDone = false;
+    if ((kind == ExportKind::Pos || kind == ExportKind::Prim) && i == exports.size() - 1)
+      exportDone = true; // Last export
+
+    m_builder->CreateIntrinsic(Intrinsic::amdgcn_exp, valueTy,
+                               {
+                                   m_builder->getInt32(target + exports[i].index), // tgt
+                                   m_builder->getInt32((1U << numValues) - 1),     // en
```

Comment:
The `en` field is a bitmask of channels. So if values is `{ x, null, z, null }`, the mask here should be 5 instead of 3.

Reply:
OK. I will correct. I previously didn't take this into account because there were no such data in the caller.